### DISCUSSION TITLE
milvus 通过源码编译安装在AutoDL 服务器 上，attu客户端工具连接不上milvus

### DISCUSSION THREAD

**Discussion by gongyafei123** [milvus-io/milvus#33364] [Q&A and General discussion]
milvus 通过源码编译安装在AutoDL 服务器 上，attu客户端工具连接不上milvus

**Comment by yhmo** [comment]
Steps to compile and run Milvus locally:

1. Enter the project root path, run this script to set environment paths.
`source ./scripts/setenv.sh`

2. Under the project root path, run this script to install the dependent libraries.
`./scripts/install_deps.sh`


3. Under the project root path, run "make" command to compile milvus.
`make`
Once it successfully compiles, you will see ./bin/milvus under the project root path.


4. Use a text tool to open the ./config/milvus.yaml, replace all the "/var/lib/milvus" to be "/tmp/milvus".
This step is to avoid the permission deny problem during milvus runtime.


5. Start the third-party services by docker, including etcd/minio.
`cd deployments/docker/dev
docker-compose up -d`

6. Back to the project root path, run this script to start a local milvus standalone.
`./scripts/start_standalone.sh`
Use command `ps -e | grep milvus` to check, you will see a milvus process running.

The runtime log will be gernerated under /tmp/standalone.log. If the server successfully starts, you will see log like this:
`---Milvus Proxy successfully initialized and ready to serve!---
[2024/03/21 18:38:57.874 +08:00] [INFO] [components/proxy.go:58] ["Proxy successfully started"]`

**Reply by gongyafei123** [reply]
有一些环境要求吧，好多会有版本冲突，建议使用哪一版本呢

**Reply by DeerEyre** [reply]
方便请教一下您如何使用milvus源码编译安装在autodl上的吗，我安装疯狂报错

**Reply by yhmo** [reply]
> 方便请教一下您如何使用milvus源码编译安装在autodl上的吗，我安装疯狂报错

我们其实也不会在每种平台上都去编译，autodl我们也没试过。

**Reply by DeerEyre** [reply]
您这边有什么推荐的外部服务器部署平台吗，因为autodl本身就是一个docker容器，无法在容器内嵌套使用docker部署milvus

**Reply by gongyafei123** [reply]
您说的是对的，后来我们通过阿里云服务器进行安装attu，autodl中只安装了milvus（源码编译安装，不建议使用这种方式，太繁琐了）

**Reply by gongyafei123** [reply]
> 方便请教一下您如何在autodl上使用milvus源码编译安装吗，我安装疯狂报错

我记得当时我们是降低了python的版本，源码编译过程中会遇到一些问题，也是通过替换报错的包版本去弄的，比较麻烦
